healthcare/healthcare/doctype/patient_encounter/patient_encounter.py

```
@@ -480,66 +480,103 @@ def create_service_request(encounter):
+
+def get_actual_quantities(item_codes, warehouse):
+	"""return {item_code: actual_qty} in the given warehouse"""
+	if not warehouse or not item_codes:
+		return {}
+
+	bins = frappe.get_all(
+		"Bin",
+		filters={"warehouse": warehouse, "item_code": ["in", item_codes]},
+		fields=["item_code", "actual_qty"],
+	)
+	return {bin.item_code: bin.actual_qty for bin in bins}
```

Ruff (0.16.0)

[error] Variable `bin` is shadowing a Python builtin

(A001)